File: app/channels/group_channel.rb
class GroupChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from "group_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(group_message)
    GroupMessage.create(body: group_message['messagebody'], user_id: group_message['user'].to_i, group_id: group_message['group'].to_i)
    ActionCable.server.broadcast 'group_channel', messagebody: group_message['messagebody']
  end
end

File: app/javascript/channels/group_channel.js
// import consumer from "./consumer"

// App.room = App.cable.subscriptions.create("GroupChannel", {
//   connected() {
//     // Called when the subscription is ready for use on the server
//   },

//   disconnected() {
//     // Called when the subscription has been terminated by the server
//   },

//   received(data) {
//     // Called when there's incoming data on the websocket for this channel
//     $('#add-message').append("<div>" + data["group_message"] + "</div>");
//   },

//   speak: (messagebody) 
//     // channelのspeakアクションにパラメータを渡す
//     @perform 'speak', {messagebody: messagebody, user: user, group: group} 
  
//   // チャットを送る
//   $(document).on('turbolinks:load', function () {
//     $(document).on 'keypress', '[data-behavior~=group_speaker]', (event) {
//     // return(Enter)が押された時
//     if event.keyCode is 13
//       //channel speakへ、event.target.valueを引数に
//       App.room.speak event.target.value, $('[data-user]').attr('data-user'), $('[data-group]').attr('data-group')
//       //inputの中身を空に
//       event.target.value = ''
//       alert()
//       event.preventDefault()
//     };
//   };
  
// });
